### PR TITLE
[dns] Introduce network scopes

### PR DESCRIPTION
**Pull request by caglar10ur**
- we now have a seperate vic-dns binary to create standalone DNS service
- context now inserts/removed truncated id and common/name into
the map of containers (using SCOPE:NAME as a key)
- dns part of the PL now iterates over scopes to map requester IP adress
and returns the SCOPE:NAME key as a response
- tether now generates /etc/resolv.conf for containers

### CODE REVIEW

File: lib/dns/dns.go [hunk 1]
```diff
@@ -284,88 +298,85 @@ Redo:
...
+	defer trace.End(trace.Begin(r.String()))
 
-	return nil
-}
+	question := r.Question[0]
 
-// ServeDNS implements the handler interface
-func (s *Server) ServeDNS(w mdns.ResponseWriter, r *mdns.Msg) {
-	defer trace.End(trace.Begin(r.String()))
+	ctx := network.DefaultContext
```
Comment (hmahmood) [inline lib/dns/dns.go]: Maybe pass in the context?

File: lib/portlayer/network/context.go [hunk 2]
```diff
@@ -480,7 +497,24 @@ func (c *Context) UnbindContainer(h *exec.Handle) error {
...
+	var key string
+	// network scoped
+	for _, e := range con.Endpoints() {
+		sname := e.Scope().Name()
+
+		key = fmt.Sprintf("%s:%s", sname, tid)
+		delete(c.containers, exec.ParseID(key))
+
+		key = fmt.Sprintf("%s:%s", sname, cname)
+		delete(c.containers, exec.ParseID(key))
```
Comment (hmahmood) [inline lib/portlayer/network/context.go]: Do we also remove from the dns cache?

Comment (caglar10ur) [inline lib/portlayer/network/context.go]: Nope, but we don't cache internal names

**Comment by hmahmood** [comment]
lgtm

**Comment by dougm** [comment]
lgtm

**Pull request merged**